### PR TITLE
2to3: add page

### PR DESCRIPTION
**Pull request by bl-ue**
Adds [2to3](https://docs.python.org/3/library/2to3.html).

- [x] The page (if new), does not already exist in the repo.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

### CODE REVIEW

File: pages/common/2to3.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# 2to3
+
+> Automated Python 2 to 3 code translation.
+> More information: <https://docs.python.org/3/library/2to3.html>.
+
+- Print changes that would be performed:
```
Comment (navarroaxel) [inline pages/common/2to3.md:6]:
```suggestion
- Display changes that would be performed:
```

File: pages/common/2to3.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+`2to3 {{file.py}}`
```
Comment (navarroaxel) [inline pages/common/2to3.md:8]:
```suggestion
`2to3 {{path/to/file.py}}`
```

File: pages/common/2to3.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+- Convert Python 2 file to Python 3:
+
+`2to3 -w {{file.py}}`
```
Comment (navarroaxel) [inline pages/common/2to3.md:12]: Use long argument format please:
```suggestion
`2to3 --write {{path/to/file.py}}`
```

File: pages/common/2to3.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+- Convert specified Python 2 features to Python 3:
+
+`2to3 -w {{file.py}} -f {{raw_input}} -f {{print}}`
```
Comment (navarroaxel) [inline pages/common/2to3.md:16]:
```suggestion
`2to3 --write {{file.py}} --fix={{raw_input}} --fix={{print}}`
```

File: pages/common/2to3.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+- Convert all features except the specified ones to Python 3:
+
+`2to3 -x {{raw_input}} -f {{print}} example.py`
```
Comment (navarroaxel) [inline pages/common/2to3.md:20]: Should the 2nd argument be `--nofix`?
```suggestion
`2to3 --nofix={{raw_input}} --fix={{print}} example.py`
```

File: pages/common/2to3.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+- Print list of available convertable features:
```
Comment (navarroaxel) [inline pages/common/2to3.md:22]:
```suggestion
- Display the list of available transformation features:
```

File: pages/common/2to3.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+`2to3 -l`
```
Comment (navarroaxel) [inline pages/common/2to3.md:24]:
```suggestion
`2to3 --list-fixes`
```

File: pages/common/2to3.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+- Convert an entire directory:
+
+`2to3 --output-dir={{python3-version/mycode}} -W -n {{python2-version/mycode}}`
```
Comment (navarroaxel) [inline pages/common/2to3.md:28]:
```suggestion
`2to3 --output-dir={{path/to/code_python3_version}} --write-unchanged-files --nobackups {{path/to/code_python2_version}}`
```

**Review by navarroaxel — CHANGES_REQUESTED** [review]

File: pages/common/2to3.md (hunk 5 above)
Comment (bl-ue) [reply pages/common/2to3.md:20]: @navarroaxel sure.  I just copied from the docs page.

**Comment by bl-ue** [comment]
@navarroaxel Thanks for the suggestions&mdash;I'll be more careful next time to use the long options.

File: pages/common/2to3.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
+# 2to3
+
+> Automated Python 2 to 3 code translation.
+> More information: <https://docs.python.org/3/library/2to3.html>.
+
+- Display changes that would be performed:
+
+`2to3 {{path/to/file.py}}`
+
+- Convert Python 2 file to Python 3:
```
Comment (schneiderl) [inline pages/common/2to3.md:10]:
```suggestion
- Convert a Python 2 file to Python 3:
```

File: pages/common/2to3.md [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+
+`2to3 --write {{path/to/file.py}}`
+
+- Convert specified Python 2 features to Python 3:
```
Comment (schneiderl) [inline pages/common/2to3.md:14]:
```suggestion
- Convert a specified Python 2 feature to Python 3:
```

**Review by schneiderl — CHANGES_REQUESTED** [review]

**Review by schneiderl — APPROVED** [review]

File: pages/common/2to3.md [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+
+`2to3 --write {{path/to/file.py}}`
+
+- Convert a specified Python 2 feature to Python 3:
+
+`2to3 --write {{file.py}} --fix={{raw_input}} --fix={{print}}`
+
+- Convert all features except the specified ones to Python 3:
+
+`2to3 --nofix={{raw_input}} --fix={{print}} example.py`
```
Comment (navarroaxel) [inline pages/common/2to3.md:20]: I just want to confirm: should this `--fix={{print}}` be `--nofix={{print}}`?

**Review by navarroaxel — CHANGES_REQUESTED** [review]

File: pages/common/2to3.md (hunk 11 above)
Comment (bl-ue) [reply pages/common/2to3.md:20]: @navarroaxel Yes.

Comment (bl-ue) [reply pages/common/2to3.md:20]: @navarroaxel Maybe it should be:
```
Convert all features except *raw_input* to Python 3:
```

Comment (navarroaxel) [reply pages/common/2to3.md:20]: Yeah, please apply.

Comment (bl-ue) [reply pages/common/2to3.md:20]: @navarroaxel done @ 17badf8d78705416d4880d793d66d97697cb51c1

File: pages/common/2to3.md [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+
+- Convert a Python 2 file to Python 3:
+
+`2to3 --write {{path/to/file.py}}`
+
+- Convert a specified Python 2 feature to Python 3:
+
+`2to3 --write {{file.py}} --fix={{raw_input}} --fix={{print}}`
+
+- Convert all features except *raw_input* the specified ones to Python 3:
```
Comment (navarroaxel) [inline pages/common/2to3.md:18]: We use back ticks here:
```suggestion
- Convert all features except `raw_input` the specified ones to Python 3:
```

**Review by navarroaxel — CHANGES_REQUESTED** [review]

File: pages/common/2to3.md (hunk 12 above)
Comment (bl-ue) [reply pages/common/2to3.md:18]: @navarroaxel I know I saw `*...*` somewhere...

Comment (bl-ue) [reply pages/common/2to3.md:18]: [`pages/common/strings.md`](https://github.com/tldr-pages/tldr/blob/master/pages/common/strings.md)

Comment (bl-ue) [reply pages/common/2to3.md:18]: [`pages/common/supervisorctl.md`](https://github.com/tldr-pages/tldr/blob/master/pages/common/supervisorctl.md)

Comment (bl-ue) [reply pages/common/2to3.md:18]: @navarroaxel `strings` uses `*`, and `supervisorctl` uses `**`.  Should I use the backtick as you've done above, and open a PR to normalize those ones that I've mentioned?

Comment (navarroaxel) [reply pages/common/2to3.md:18]: Well... We should improve our standard. Leave the * then.

**Review by navarroaxel — APPROVED** [review]

File: pages/common/2to3.md (hunk 12 above)
Comment (bl-ue) [reply pages/common/2to3.md:18]: @navarroaxel should I open a PR to correct those?  I can correct the standard, too :)

Comment (bl-ue) [reply pages/common/2to3.md:18]: @owenvoke WDYT?

Comment (owenvoke) [reply pages/common/2to3.md:18]: For the `supervisorctl`, I don't think it needs the bold text (and as it's not part of the standard, it looks weird).

As for the italics in `strings`, I think we could just remove it there too, and replace the "length" with an example number. e.g.

```
- Limit results to strings that are at least 10 characters long:

`strings -n {{10}} {{file}}`
```

Comment (bl-ue) [reply pages/common/2to3.md:18]: >and replace the "length" with an example number

Good idea.  That must be the bit that I was missing.  Thank you!

**Pull request merged**